OSOE-353: Add duplicate SQL query detector in Lombiq.UITestingToolbox

dministro opened the pull request:
[OSOE-353](https://lombiq.atlassian.net/browse/OSOE-353)

[OSOE-353]: https://lombiq.atlassian.net/browse/OSOE-353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dministro committed:
Adding negative tests
Adjusting tests
Updating submodule:
 - Lombiq.UITestingToolbox: merge from dev

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodules:
 - Lombiq.UITestingToolbox: Implementing required exception constructors

dministro committed:
Adjusting spelling configuration
Updating submodules:
 - Lombiq.HelpfulLibraries: fast-forward
 - Lombiq.OrchardCoreApiClient: fast-forward
 - Lombiq.HelpfulExtensions: fast-forward
 - Lombiq.Privacy: fast-forward
 - Lombiq.TrainingDemo: fast-forward
 - Lombiq.VueJs: fast-forward
 - Lombiq.Npm.Targets: fast-forward
 - Lombiq.UITestingToolbox: Adjusting tests Fixing typos
 - Lombiq.Analyzers.PowerShell: fast-forward
 - Lombiq.GitHub.Actions: Adjusting allow.txt

dministro committed:
Updating submodules:
 - Lombiq.UITestingToolbox: Fixing typo
 - Lombiq.GitHub.Actions: Adjusting allow.txt

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

Piedone commented on test/Lombiq.OSOCE.Tests.UI/Tests/RetrievalDataTablesTests.cs:
```diff
@@ -18,5 +18,11 @@ public BehaviorDataTablesTests(ITestOutputHelper testOutputHelper)
         ExecuteTestAfterSetupAsync(
             context => context.TestDataTableRecipeDataAsync(),
-            browser);
+            browser,
```
You didn't do it but the file and class names are different. Rename the file.

Piedone commented on test/Lombiq.OSOCE.Tests.UI/Tests/ModuleTests/BehaviorChartJsTests.cs:
```diff
@@ -18,5 +18,11 @@ public BehaviorChartJsTests(ITestOutputHelper testOutputHelper)
+            configuration =>
+            {
+                configuration.CounterConfiguration.Running.DbReaderReadThreshold = 404;
+                configuration.CounterConfiguration.Running.DbReaderReadPerNavigationThreshold = 70;
```
This and `BehaviorDataTablesTests` should rather specify thresholds for specific pages, see https://github.com/Lombiq/UI-Testing-Toolbox/pull/216#discussion_r1004911693.

Piedone commented:
On the next commit, UI testing will break if you don't merge from `dev` beforehand.

dministro commented:
> On the next commit, UI testing will break if you don't merge from `dev` beforehand.

#258 , Thank you.

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Moving and renaming test classes

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Adding better exclusion and defaults

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Fast forwarding submodules

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Update submodules after merge

dministro committed:
Updating submodule:
 - Lombiq.GitHub.Actions: Allowing ICompiled

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodules:
 - Lombiq.HelpfulLibraries: fast forward
 - Lombiq.DataTables: fast forward
 - Lombiq.Hosting.Tenants: fast forward
 - Lombiq.BaseTheme: fast forward
 - Lombiq.Analyzers: fast forward
 - Lombiq.Analyzers.PowerShell: fast forward
[no ci]

dministro committed:
dummy

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodules after merge

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Fixing analyzer errors

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodules
[no ci]

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodules

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Fixes after merge

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Fixes after merge

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Addressing "S4457: Split this method into two..." analyzer error

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodules

dministro committed:
Reverting pnpm-lock.yaml
[no ci]

dministro committed:
Adjusting tests to utilize per page configuration
Updating submodule:
 - Lombiq.UITestingToolbox: Adding per page configuration implementation

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353
[no ci]

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodule:

 - Lombiq.UITestingToolbox: Fixing analyzer violations

dministro committed:
Fixing failing UI tests
Updating submodule:
 - Lombiq.UITestingToolbox: Adding ProbedSqliteConnection

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Removing tests
Updating submodule:
 - Lombiq.UITestingToolbox: merge from dev

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
[no ci]

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Adding comment

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Fix spelling

dministro committed:
Reverting changes

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353
[no ci]

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353
[noci]

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Fixes after merge
[noci]

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Fixing analyzer violations

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Fixing analyzer violations again

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
[noci]

dministro committed:
Updating submodules

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353
[noci]

dministro committed:
Updating submodule:
- Lombiq.UITestingToolbox: Spelling

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Fixing analyzer warnings

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
submodules

dministro committed:
Submodule

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Reverts #73983ef2e70eec272c92b3016268db8d0e0abacf

dministro commented on NuGetTest/src/Lombiq.OSOCE.NuGet.Web/Lombiq.OSOCE.NuGet.Web.csproj:
```diff
@@ -11,7 +11,7 @@
     <ExecDotnetPostcleanCommand>false</ExecDotnetPostcleanCommand>
     <ExecDotnetPrebuildCommand>false</ExecDotnetPrebuildCommand>
-    <OrchardCoreVersion>2.1.7</OrchardCoreVersion>
+    <OrchardCoreVersion>2.1.6</OrchardCoreVersion>
```
There is a breaking change in Yesql, which is a transient dependency. The `ISession.DetachAll` and `ISession.ResetAsync` were introduced.
Pointless and risky to upgrade to 2.1.7 until the submodules are on 2.1.0.

This reverts the change introduced by 73983ef2e70eec272c92b3016268db8d0e0abacf.

Piedone replied:
What is the breaking change that you're referring to? I assume not the introduction of two new methods, since that's not breaking.

We keep the Web projects on the latest OC, and submodules on .0 for a good reason: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions?tab=readme-ov-file#net-and-orchard-core-version-support-policy

dministro replied:
Introducing new methods on an interface which is not implemented in a dependency which is references a different version is a breaking change and results in a `TypeLoadException`.
See: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/16155545840/job/45611756550
![0-AfterPageChange](https://github.com/user-attachments/assets/8df1acfc-1d2e-4bc6-af72-7298d450134e)

I can uprade to 2.1.7 in `Lombiq.Tests.UI`, but it is a big impact, because I have to upgrade OC modules in referenced modules, so ...
What is the expectation? keep `Lombiq.Tests.UI` on 2.1.0 or upgrade to 2.1.7(under a different PR because of the impact)?

Piedone replied:
Why do you need to touch anything related to this in the first place? What's at the root of all of this?

dministro replied:
The `TypeLoadException` (see: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/pull/247#discussion_r2196071215).

Piedone replied:
I see the comment two comments up :). My question is, what's before that? Link to the code that requires you to change the OC version.

dministro replied:
Here is the `ISession` implementation(`Lombiq.Tests.UI.Services.Counters.SessionProbe`) in UITT: https://github.com/Lombiq/UI-Testing-Toolbox/blob/a547aab779d95de56bd0fe78d3de8658bc25044f/Lombiq.Tests.UI/Services/Counters/SessionProbe.cs#L12

Piedone replied:
I see. Keep the Web projects on 2.1.7. Update UITT to 2.1.7 too, but add it as a temporary exception here: https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/workflows/build-and-test-orchard-core.yml#L211 With a comment that explains why it's there, that includes the phrase "Orchard Core upgrade".

dministro replied:
I've tried it already, and it ends in a version downgrade compile time warning (/warningaserror) because of HL (if I remember correctly, it's on 2.1.0)

I'll give it a try. If the version downgrade warning is still present, should I upgrade OC dependencies in HL(and so on) or keep the web project on 2.1.6?

Piedone replied:
Since UITT is used is many projects where the Web project is on 2.1.7, downgrading that to 2.1.6 is not a viable option.

I checked, and only `Lombiq.HelpfulExtensions.Tests.UI` is affected, where the current `OrchardCore.ContentManagement.Abstractions` that breaks this isn't actually needed and can be removed.

Piedone replied:
Sorry, the reference is needed but can be changed to 2.1.7 in the same manner.

dministro replied:
After upgrading OC modules to 2.1.7 in HL:
```
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Core -> OrchardCore.Media 2.1.0 -> OrchardCore.ContentManagement.Display 2.1.7 -> OrchardCore.DisplayManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Core -> OrchardCore.DisplayManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ContentManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Core -> OrchardCore.Media 2.1.0 -> OrchardCore.ContentManagement.GraphQL 2.1.7 -> OrchardCore.ContentManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Core -> OrchardCore.ContentManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Core -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.ResourceManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Core -> OrchardCore.ResourceManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Module.Targets from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Module.Targets (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Module.Targets (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Media 2.1.0 -> OrchardCore.Apis.GraphQL.Abstractions 2.1.7 -> OrchardCore.DisplayManagement.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.DisplayManagement.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Media 2.1.0 -> OrchardCore.DisplayManagement.Liquid 2.1.7 -> OrchardCore.ResourceManagement.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.ResourceManagement.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Deployment.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.Deployment.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Deployment.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Recipes.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.Recipes.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Recipes.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Module.Targets from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.Tenants.EmailQuotaManagement -> Lombiq.HelpfulExtensions -> OrchardCore.Module.Targets (>= 2.1.7) 
 Lombiq.Hosting.Tenants.EmailQuotaManagement -> OrchardCore.Module.Targets (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Infrastructure.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.Tenants.EmailQuotaManagement -> OrchardCore.Email 2.1.0 -> OrchardCore.Admin.Abstractions 2.1.7 -> OrchardCore.Infrastructure.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.Tenants.EmailQuotaManagement -> OrchardCore.Infrastructure.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.Tenants.EmailQuotaManagement -> OrchardCore.Email 2.1.0 -> OrchardCore.Navigation.Core 2.1.7 -> OrchardCore.DisplayManagement (>= 2.1.7) 
 Lombiq.Hosting.Tenants.EmailQuotaManagement -> OrchardCore.DisplayManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.Tenants.EmailQuotaManagement -> Lombiq.HelpfulLibraries.OrchardCore -> OrchardCore.Markdown 2.1.7 -> OrchardCore.ResourceManagement (>= 2.1.7) 
 Lombiq.Hosting.Tenants.EmailQuotaManagement -> OrchardCore.ResourceManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Workflows from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Privacy.Samples -> Lombiq.HelpfulExtensions -> OrchardCore.Workflows (>= 2.1.7) 
 Lombiq.Privacy.Samples -> OrchardCore.Workflows (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Module.Targets from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Privacy.Samples -> Lombiq.HelpfulExtensions -> OrchardCore.Module.Targets (>= 2.1.7) 
 Lombiq.Privacy.Samples -> OrchardCore.Module.Targets (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> OrchardCore.Media 2.1.0 -> OrchardCore.ContentManagement.Display 2.1.7 -> OrchardCore.DisplayManagement (>= 2.1.7) 
 Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> OrchardCore.DisplayManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ContentManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> OrchardCore.Media 2.1.0 -> OrchardCore.ContentManagement.GraphQL 2.1.7 -> OrchardCore.ContentManagement (>= 2.1.7) 
 Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> OrchardCore.ContentManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.ResourceManagement (>= 2.1.7) 
 Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> OrchardCore.ResourceManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> OrchardCore.Media 2.1.0 -> OrchardCore.ContentManagement.Display 2.1.7 -> OrchardCore.DisplayManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> OrchardCore.DisplayManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ContentManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> OrchardCore.Media 2.1.0 -> OrchardCore.ContentManagement.GraphQL 2.1.7 -> OrchardCore.ContentManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> OrchardCore.ContentManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.ResourceManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> OrchardCore.ResourceManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ContentManagement.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.HelpfulExtensions.Tests.UI -> Lombiq.HelpfulExtensions -> OrchardCore.MetaWeblog.Abstractions 2.1.7 -> OrchardCore.ContentManagement.Abstractions (>= 2.1.7) 
 Lombiq.HelpfulExtensions.Tests.UI -> OrchardCore.ContentManagement.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Module.Targets from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Module.Targets (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Module.Targets (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Media 2.1.0 -> OrchardCore.Apis.GraphQL.Abstractions 2.1.7 -> OrchardCore.DisplayManagement.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.DisplayManagement.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Media 2.1.0 -> OrchardCore.DisplayManagement.Liquid 2.1.7 -> OrchardCore.ResourceManagement.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.ResourceManagement.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Deployment.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.Deployment.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Deployment.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Recipes.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.Recipes.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Recipes.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> OrchardCore.ContentManagement 2.1.0 -> OrchardCore.Infrastructure.Abstractions 2.1.7 -> OrchardCore.ResourceManagement.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.ResourceManagement.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Module.Targets from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Module.Targets (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Module.Targets (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> OrchardCore.ContentManagement 2.1.0 -> OrchardCore.ContentTypes.Abstractions 2.1.7 -> OrchardCore.ContentManagement.Display 2.1.7 -> OrchardCore.DisplayManagement (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> OrchardCore.DisplayManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Media 2.1.0 -> OrchardCore.Apis.GraphQL.Abstractions 2.1.7 -> OrchardCore.DisplayManagement.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.DisplayManagement.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ContentManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Media 2.1.0 -> OrchardCore.ContentManagement.GraphQL 2.1.7 -> OrchardCore.ContentManagement (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> OrchardCore.ContentManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Deployment.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.Deployment.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Deployment.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Recipes.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.Recipes.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Recipes.Abstractions (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.ResourceManagement (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Tests.Theme -> OrchardCore.ResourceManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Workflows from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Privacy.Tests.UI -> Lombiq.Privacy.Samples -> Lombiq.HelpfulExtensions -> OrchardCore.Workflows (>= 2.1.7) 
 Lombiq.Privacy.Tests.UI -> Lombiq.Privacy.Samples -> OrchardCore.Workflows (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Native.Samples -> OrchardCore.Admin 2.1.0 -> OrchardCore.Navigation.Core 2.1.7 -> OrchardCore.DisplayManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Native.Samples -> Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> OrchardCore.DisplayManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ContentManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Native.Samples -> Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> OrchardCore.Media 2.1.0 -> OrchardCore.ContentManagement.GraphQL 2.1.7 -> OrchardCore.ContentManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Native.Samples -> Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> OrchardCore.ContentManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Native.Samples -> Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.ResourceManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Native.Samples -> Lombiq.BaseTheme.Native -> Lombiq.BaseTheme.Core -> OrchardCore.ResourceManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.DisplayManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Samples -> OrchardCore.Admin 2.1.0 -> OrchardCore.Navigation.Core 2.1.7 -> OrchardCore.DisplayManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Samples -> Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> OrchardCore.DisplayManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ContentManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Samples -> Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> OrchardCore.Media 2.1.0 -> OrchardCore.ContentManagement.GraphQL 2.1.7 -> OrchardCore.ContentManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Samples -> Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> OrchardCore.ContentManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.ResourceManagement from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.BaseTheme.Samples -> Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.ResourceManagement (>= 2.1.7) 
 Lombiq.BaseTheme.Samples -> Lombiq.BaseTheme -> Lombiq.BaseTheme.Core -> OrchardCore.ResourceManagement (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Workflows from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI -> Lombiq.Privacy.Tests.UI -> Lombiq.Privacy.Samples -> Lombiq.HelpfulExtensions -> OrchardCore.Workflows (>= 2.1.7) 
 Lombiq.Hosting.Azure.ApplicationInsights.Tests.UI -> Lombiq.Privacy.Tests.UI -> Lombiq.Privacy.Samples -> OrchardCore.Workflows (>= 2.1.0)
Warning As Error: Detected package downgrade: OrchardCore.Deployment.Abstractions from 2.1.7 to 2.1.0. Reference the package directly from the project to select a different version. 
 Lombiq.Hosting.MediaTheme.Bridge.Tests.UI -> Lombiq.Hosting.MediaTheme.Bridge -> Lombiq.HelpfulExtensions -> OrchardCore.Contents 2.1.7 -> OrchardCore.Deployment.Abstractions (>= 2.1.7) 
 Lombiq.Hosting.MediaTheme.Bridge.Tests.UI -> Lombiq.Hosting.MediaTheme.Bridge -> OrchardCore.Deployment.Abstractions (>= 2.1.0)
```

This is what I mean under "...and so on"

Piedone replied:
You don't need to update Helpful Libraries. The only other affected project is `Lombiq.HelpfulExtensions.Tests.UI`.

dministro replied:
Adding `OrchardCore.ContentManagement.Abstractions` 2.1.7 as a direct reference to HL looks better, but I can't see the impact.

dministro replied:
Domino effect: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/16206856185/job/45759029743?pr=247

dministro replied:
> You don't need to update Helpful Libraries. The only other affected project is `Lombiq.HelpfulExtensions.Tests.UI`.

Sorry, it was a typo. I mean HE

Piedone replied:
But that's not necessary. Why would you do that?

dministro replied:
After removing `OrchardCore.ContentManagement.Abstractions` direct reference from `Lombiq.HelpfulExtensions.csproj`, the dotnet consolidation is still failing.

https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/16206960865/job/45759363976?pr=247
```
----------------------------
OrchardCore.ContentFields
----------------------------
Lombiq.ChartJs.Samples - 2.1.0
Lombiq.ContentEditors - 2.1.0
Lombiq.DataTables.Samples - 2.1.0
Lombiq.HelpfulExtensions - 2.1.0
Lombiq.HelpfulLibraries.OrchardCore - 2.1.0
Lombiq.HelpfulLibraries.Samples - 2.1.0
Lombiq.JsonEditor - 2.1.0
Lombiq.TrainingDemo - 2.1.0
Lombiq.VueJs.Samples - 2.1.0
Lombiq.Walkthroughs - 2.1.0
Lombiq.Tests.UI - 2.1.7
----------------------------
OrchardCore.Abstractions
----------------------------
Lombiq.HelpfulLibraries.OrchardCore - 2.1.0
Lombiq.Hosting.Tenants.IdleTenantManagement - 2.1.0
Lombiq.SetupExtensions - 2.1.0
Lombiq.Tests.UI.Shortcuts - 2.1.0
Lombiq.Tests.UI - 2.1.7
----------------------------
OrchardCore.ContentManagement.Abstractions
----------------------------
Lombiq.HelpfulLibraries.OrchardCore - 2.1.0
Lombiq.OrchardCoreApiClient - 2.1.0
Lombiq.HelpfulExtensions.Tests.UI - 2.1.7
```

Piedone replied:
I elaborated what I suggest doing above, what is not this. Please just try that. (Not remove but change the HE reference, exclude the two projects from the consolidation.)

dministro commented on src/Lombiq.OSOCE.Web/Lombiq.OSOCE.Web.csproj:
```diff
@@ -4,7 +4,7 @@
     <TieredCompilation>true</TieredCompilation>
     <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
-    <OrchardCoreVersion>2.1.7</OrchardCoreVersion>
+    <OrchardCoreVersion>2.1.6</OrchardCoreVersion>
```
There is a breaking change in Yesql, which is a transient dependency. The `ISession.DetachAll` and `ISession.ResetAsync` were introduced.
Pointless and risky to upgrade to 2.1.7 until the submodules are on 2.1.0.

This reverts the change introduced by 73983ef2e70eec272c92b3016268db8d0e0abacf.

dministro committed:
Updating submodule

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-353

dministro committed:
Reverting OC version changes
Updating submodules

dministro committed:
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Branch references
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Updating submodule

dministro committed:
Excluding additional project files

dministro committed:
Excluding again with the correct parameter name

dministro committed:
Removing quotes

dministro committed:
Branch reference

Piedone committed:
Formatting

[skip ci]

Piedone committed:
Disabling Windows .NET consolidation check

[skip ci]

Piedone committed:
Also for the NuGet build

[skip ci]

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-353

# Conflicts:
#	test/Lombiq.UITestingToolbox

Piedone committed:
Grammar/clarifying docs, code styling

Piedone reviewed (changes requested):
Release a NuGet alpha and update the NuGetTest solution with it: https://lombiq.atlassian.net/wiki/spaces/DEV/pages/786857987/Managing+releases+of+open-source+projects

Piedone commented:
Somebody else will do this.
